src/models/polar/constraints.jl

```diff
@@ -207,6 +207,50 @@ function flow_constraints(polar::PolarForm, cons, buffer)
     wait(ev)
     return
 end
+
+function flow_constraints_grad(polar::PolarForm, cons, buffer, reduction::Function=sum)
+    function accumulate_view!(x, vx, indices)
+        ids = unique(indices)
+        # This is parallelizable 
+        for id in ids
+            for (j, idx) in enumerate(indices)
+                if id == idx 
+                    x[id] += vx[j]
+                end
+            end
+        end
+    end
+    nlines = PS.get(polar.network, PS.NumberOfLines())
+    nbus = PS.get(polar.network, PS.NumberOfBuses())
+    f = polar.topology.f_buses
+    t = polar.topology.t_buses
+    fr_vmag = buffer.vmag[f]
+    to_vmag = buffer.vmag[t]
+    fr_vang = buffer.vang[f]
+    to_vang = buffer.vang[t]
+    function lumping(fr_vmag, to_vmag, fr_vang, to_vang)
+        Δθ = fr_vang .- to_vang
+        cosθ = cos.(Δθ)
+        sinθ = sin.(Δθ)
+        cons = branch_flow_kernel_zygote(
+            polar.topology.yff_re, polar.topology.yft_re, polar.topology.ytf_re, polar.topology.ytt_re,
+            polar.topology.yff_im, polar.topology.yft_im, polar.topology.ytf_im, polar.topology.ytt_im,
+            fr_vmag, to_vmag,
+            cosθ, sinθ
+        )
+        return reduction(cons)
+    end
+    grad = Zygote.gradient(lumping, fr_vmag, to_vmag, fr_vang, to_vang)
+    gvmag = similar(buffer.vmag)
+    gvang = similar(buffer.vang)
```

Comment:
This is not a priority for this PR, but I would maybe add a `TODO` to tell that we would need to move `gvmag` and `gvang` in a buffer.